2226/2/raper.rb

```diff
@@ -0,0 +1,50 @@
+# This class is used for single raper methods
+class Raper
+  attr_reader :raper_name
+
+  SWEAR_WORDS = YAML.load_file('config.yml')['SWEAR_WORDS_ARRAY'].join('|')
+
+  def initialize(raper_name)
+    @raper_name = raper_name
+  end
+
+  def avg_number_of_battles
+    DataStorage.show_all_data.sum do |file_name, _|
+      file_name.scan(@raper_name).size
+    end
+  end
+
+  def number_of_battles
+    avg_number_of_battles / 2
+  end
+
+  def number_of_swear_words
+    DataStorage.show_all_data.inject(0) do |counter, (file_name, file_content)|
+      counter += file_content.scan(/#{SWEAR_WORDS}/).size if file_name.match(@raper_name)
+      counter
+    end
+  end
+
+  def average_number_swearing_words_in_battle
+    number_of_swear_words.fdiv(number_of_battles)
+  end
+
+  def number_of_words_in_rounds
+    DataStorage.show_all_data.inject(0) do |counter, (file_name, file_content)|
+      counter += file_content.split(/ [а-яА-ЯёЁ*]+/).size if file_name.match(@raper_name)
```

Comment:
`file_name.match(@raper_name) `
is repeated everywhere. It has to be a separate method probably. But actually, you don't need it only once.